.github/workflows/tp.yml

```diff
@@ -18,10 +18,10 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-node-
 
-      - name: Use Node.js 14.x
+      - name: Use Node.js 15.x
         uses: actions/setup-node@v1
         with:
-          node-version: 14.x
+          node-version: 15.x
```

Comment:
I agree that upgrading to `node v15.x.x` (which embeds `npm@7`) did solve the incompatibility issue with a `package-lock.json` generated with `npm@6` , but I think it's better to stick to `node v14.x.x` to avoid further issues.

In fact, Angular is using only [active or maintenance LTS](https://angular.io/guide/setup-local#prerequisites) versions of Node. Some other tools (like schematics) might break if using a non-supported node engine (I had the case recently!)

To solve the build issue due to the embedded `npm@6` in node@14 , we could instead update `npm` **in an additional step of the pipeline**, as such:

```
      - name: Install dependencies
        run: npm install -g npm@latest
        run: npm ci
```

Reply:
I think in general project you are totally right, but in our case it's a formation purpose project, schools/institutes students will maybe use node 15/16 to run ours exercises so I think we can go to the latest node version to at least know quickly what will probably fail. Maybe we can use a strategy matrix to run on both version ?

Reply:
I will fixe node version because node 15 and node 16 is not actilvely support by Angular Team